src/settings_internal.h

```diff
@@ -19,21 +19,21 @@
  * @see SettingDescBase
  */
 enum SettingDescType : byte {
-	/* 4 bytes allocated a maximum of 16 types for GenericType */
-	SDT_BEGIN       = 0,
 	SDT_NUMX        = 0, ///< any number-type
 	SDT_BOOLX       = 1, ///< a boolean number
 	SDT_ONEOFMANY   = 2, ///< bitmasked number where only ONE bit may be set
 	SDT_MANYOFMANY  = 3, ///< bitmasked number where MULTIPLE bits may be set
 	SDT_INTLIST     = 4, ///< list of integers separated by a comma ','
 	SDT_STDSTRING   = 6, ///< \c std::string
-	SDT_END,
-	/* 9 more possible primitives */
 };
 
-
-enum SettingGuiFlag : uint16 {
-	/* 1 byte allocated for a maximum of 8 flags
+/**
+ * Set of flags that describes what specialized behaviour should be performed
+ * on the setting, e.g. when things may only be changed when not in a network
+ * game, or settings that only make sense for new games.
+ */
+enum SettingFlag : uint16 {
```

Comment:
Not sure I agree with you here. You say `the rest is usually not even used by the GUI, so it should not have GUI is its name.`. From what I can tell, this is simply not true.

Most of these settings influence if a setting can be edited in the current state of the game. This is mostly meant to prevent the user from changing it, and this is exactly what it does. It disables the buttons in the settings GUI depending on the state of the game.
Some other settings, like `0ISDISABLED` give a nice label to the value `0`, which is also a GUI-only thingy.

Of course this also needs some validation in the rest of the code, but one is a result of the other. From what I can tell, most of these settings are only also used in `CmdChangeSetting`, outside of GUI usage. `0ISDISABLED` and `MULTISTRING` are also used during clamping, but that is just to allow users to edit the value directly (instead of the `<` and `>` buttons)

So I am not sure about this change; either I am missing something in the story, or this doesn't add up :) The setting-system is far from easy to understand (understatement of the century), so I might be missing something completely here :D